test/reference64/typical_merge_gallop.txt

```diff
@@ -33,316 +33,310 @@ quote
     phase_stop = min(y_lvl.shape, y_lvl_i1, x_lvl_i1)
     if phase_stop >= 1
         i = 1
+        if y_lvl.idx[y_lvl_q] < 1
+            y_lvl_q = Finch.scansearch(y_lvl.idx, 1, y_lvl_q, y_lvl_q_stop - 1)
+        end
+        if x_lvl.idx[x_lvl_q] < 1
+            x_lvl_q = Finch.scansearch(x_lvl.idx, 1, x_lvl_q, x_lvl_q_stop - 1)
+        end
         while i <= phase_stop
-            if y_lvl.idx[y_lvl_q] < i
```

Comment:
I notice that the seek has moved. Do we think this new code is improved?

Reply:
I think the new code will be faster than the previous one since it avoids calling a `scansearch` inside the while loop everytime. What I'm worried is : is lifting `scansearch` for jumper always a safe transformation?

Reply:
that's not always true, the stepper might be skipped (e.g. multiply by zero)